README.md

```diff
@@ -1,43 +1,420 @@
+   - **Mac/Linux users**: One terminal in `frontend` folder, one in `backend` folder
+   - **Windows users**: WSL terminal for `backend` folder, WSL or Windows terminal for `frontend` folder
+
+   ```bash
+   # In both terminal windows
+   export APOS_EXTERNAL_FRONT_KEY=my-secret-key
+   ```
+
+   The `astro.config.mjs` file uses default values, but if running the backend on a different port, also set:
+   ```bash
+   export APOS_HOST=your-backend-url
+   ```
+
+3. **Start development servers**
+
+   ```bash
+   # Terminal 1 - Backend (use WSL on Windows)
+   cd backend && npm run dev
+
+   # Terminal 2 - Frontend
+   cd frontend && npm run dev
+   ```
+
+> **Note:** Astro is less stringent about project setup in development mode. Before deployment, run `npm run build` followed by `npm run preview` in the `frontend` folder to test production behavior. We don't recommend using the root `npm run serve-frontend` script during development - it's used for Apostrophe hosting.
+
+Visit `http://localhost:4321` and start building! 🎉
+
+---
+
+## 🏗️ Architecture Overview
+
+
+### How It Works
+This project utilizes ApostropheCMS as a headless backend with Astro as a frontend. What sets this apart from typical headless setups is the [apostrophe-astro](https://github.com/apostrophecms/apostrophe-astro) package in the Astro frontend project. This enables full use of the ApostropheCMS Admin UI, including in-context editing, while largely automating content fetching from the backend without writing REST API calls.
+
+### Project Structure
+```
+├── backend/               # ApostropheCMS application
+│   ├── modules/           # Custom modules (pages, pieces, widgets)
+│   ├── app.js             # Main configuration
+│   └── ...
+├── frontend/              # Astro application
+│   ├── src/
+│   │   ├── pages/         # Single [...slug].astro route
+│   │   ├── templates/     # Page templates
+│   │   ├── widgets/       # Widget templates
+│   │   └── components/    # Astro components
+│   ├── astro.config.mjs   # Astro configuration
+│   └── ...
+└── README.md              # This file
+└── package.json.          # Whole project scripts
+```
+
+### For ApostropheCMS Developers
+
+If you've worked with ApostropheCMS previously, the backend should look familiar. Custom modules for pages, pieces, and widgets are in the `modules` folder, with core module configuration in `modules/@apostrophecms`.
+
+**What stays the same:**
+- Module registration in `app.js`
+- Page types added to `modules/@apostrophecms/page/index.js`
+- Most [module configuration settings](https://docs.apostrophecms.org/reference/module-api/module-overview.html#module-configuration) for Admin UI, request routing, and MongoDB interaction
+
+**Key differences:**
+- **No frontend code in modules** - Client-side JavaScript, styling, and templates go in the Astro project instead
```

Comment:
Technically this sentence is correct but it kinda reads like it's only talking about client side things. I would say "stylesheets, templates (implemented as Astro components), and client-side javascript"

Reply:
Changed.